modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IncrementalSnapshotVerificationTask.java

```diff
@@ -0,0 +1,441 @@
+                                    activeDhtTxs.add(txRec.writeVersion());
+                            }
+                        }
+                    }
+                    else if (txRec.state() == TransactionState.COMMITTED) {
+                        activeDhtTxs.remove(txRec.writeVersion());
+
+                        Set<Short> participatingNodes = txPrimParticipatingNodes.remove(txRec.nearXidVersion());
+
+                        // Legal cases:
+                        // 1. This node is a transaction near node, but not primary or backup node.
+                        // 2. This node participated in the transaction multiple times (e.g., primary for one key and backup for other key).
+                        // 3. A transaction is included into previous incremental snapshot.
+                        if (participatingNodes == null)
+                            return;
+
+                        calcTxHash.accept(txRec.nearXidVersion(), participatingNodes);
+                    }
+                    else if (txRec.state() == TransactionState.ROLLED_BACK) {
+                        activeDhtTxs.remove(txRec.writeVersion());
+                        txPrimParticipatingNodes.remove(txRec.nearXidVersion());
+                    }
+                });
+
+                // All active transactions that didn't log COMMITTED or ROLL_BACK records are considered committed.
+                // It is possible as incremental snapshot started after transaction left IgniteTxManager#activeTransactions() collection,
+                // but completed before the final TxRecord was written.
+                for (Map.Entry<GridCacheVersion, Set<Short>> tx: txPrimParticipatingNodes.entrySet())
+                    calcTxHash.accept(tx.getKey(), tx.getValue());
+
+                Map<Object, TransactionsHashRecord> txHashRes = nodesTxHash.entrySet().stream()
+                    .map(e -> new TransactionsHashRecord(
+                        consId,
+                        blt.compactIdMapping().get(e.getKey()),
+                        e.getValue().hash
+                    ))
+                    .collect(Collectors.toMap(
+                        TransactionsHashRecord::remoteConsistentId,
+                        Function.identity()
+                    ));
+
+                Map<PartitionKeyV2, PartitionHashRecordV2> partHashRes = partMap.entrySet().stream()
+                    .collect(Collectors.toMap(
+                        Map.Entry::getKey,
+                        e -> new PartitionHashRecordV2(
+                            e.getKey(),
+                            false,
+                            consId,
+                            e.getValue().hash,
+                            e.getValue().verHash,
+                            null,
+                            0,
+                            null)
+                    ));
+
+                return new IncrementalSnapshotVerificationTaskResult(
```

Comment:
Let's locally log verification results here on info level.
Including number of dataEntry (proc.applied) and transactions